e2e/tests/functional/styling.e2e.spec.js

```diff
@@ -0,0 +1,293 @@
+    flexibleLayout = await createDomainObjectWithDefaults(page, {
+      type: 'Flexible Layout',
+      name: 'Flexible Layout'
+    });
+
+    // Create a Stacked Plot and attach to the Flexible Layout
+    stackedPlot = await createDomainObjectWithDefaults(page, {
+      type: 'Stacked Plot',
+      name: 'Stacked Plot',
+      parent: flexibleLayout.uuid
+    });
+
+    // Create two SWGs and attach them to the Stacked Plot
+    sineWaveObject = await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator',
+      name: 'Sine Wave Generator 1',
+      parent: stackedPlot.uuid
+    });
+
+    sineWaveObject2 = await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator',
+      name: 'Sine Wave Generator 2',
+      parent: stackedPlot.uuid
+    });
+  });
+
+  test('styling the flexible layout properly applies the styles to all containers', async ({
+    page
+  }) => {
+    // Values to set styles to
+    const backgroundColor = '#5b0f00';
+    const textColor = '#e6b8af';
+
+    // Directly navigate to the flexible layout
+    await page.goto(flexibleLayout.url, { waitUntil: 'domcontentloaded' });
+
+    // Edit Flexible Layout
+    await page.getByLabel('Edit').click();
+
+    // Select styles tab
+    await page.getByRole('tab', { name: 'Styles' }).click();
+
+    // Set styles using setStyles function
+    await setStyles(page, backgroundColor, textColor, page.locator('.c-fl__container-holder'));
+
+    // Check styles using checkStyles function
+    await checkStyles(
+      page,
+      hexToRGB(backgroundColor),
+      hexToRGB(textColor),
+      page.locator('.c-fl__container-holder')
+    );
+
+    // Save Flexible Layout
+    await page.getByRole('button', { name: 'Save' }).click();
+    await page.getByRole('listitem', { name: 'Save and Finish Editing' }).click();
+
+    // Reload page
+    await page.reload({ waitUntil: 'domcontentloaded' });
+
+    // Check styles using checkStyles function
+    await checkStyles(
+      page,
+      hexToRGB(backgroundColor),
+      hexToRGB(textColor),
+      page.locator('.c-fl__container-holder')
+    );
+  });
+
+  test('styling a child object of the flexible layout properly applies that style to only that child', async ({
+    page
+  }) => {
+    const backgroundColor = '#5b0f00';
+    const textColor = '#e6b8af';
+    // Directly navigate to the flexible layout
+    await page.goto(flexibleLayout.url, { waitUntil: 'domcontentloaded' });
+
+    // Edit Flexible Layout
+    await page.getByLabel('Edit').click();
+
+    // Select styles tab
+    await page.getByRole('tab', { name: 'Styles' }).click();
+
+    // Set styles using setStyles function
+    await setStyles(page, backgroundColor, textColor, page.getByLabel('Stacked Plot Frame'));
+
+    // Check styles using checkStyles function
+    await checkStyles(
+      page,
+      hexToRGB(backgroundColor),
+      hexToRGB(textColor),
+      page.getByLabel('Stacked Plot Frame')
```

Comment:
but not here. I'm thinking is something fundamentally wrong in the way we are selecting elements for comparison.